### PR TITLE
Add GCG

### PR DESCRIPTION
**Pull request by jurgen-lentz**

### CODE REVIEW

File: .github/workflows/build_binaries.yml [hunk 1]
```diff
@@ -54,6 +59,7 @@ jobs:
           - name: Print inputs
             run: |
               echo "SCIP version: ${{github.event.inputs.scip_version}}"
+              echo "SCIP version: ${{github.event.inputs.gcg_version}}"
```
Comment (mmghannam) [inline .github/workflows/build_binaries.yml:62]:
```suggestion
              echo "GCG version: ${{github.event.inputs.gcg_version}}"
```

**Comment by mmghannam** [comment]
Hi @jurgen-lentz, thank you! from a first look it looks good to me. I suggest we merge this first to a `gcg` branch on this repo before merging to main. So that we'd be able to trigger the workflow and test the binaries first.

**Comment by mmghannam** [comment]
Just triggered it, the run's result should be here https://github.com/scipopt/scipoptsuite-deploy/actions/runs/9623870769

**Comment by jurgen-lentz** [comment]
Could you maybe push into the ```main``` branch and make a new release with it ```v0.4.1```? I need it to build pygcgopt.

**Comment by mmghannam** [comment]
> Could you maybe push into the `main` branch and make a new release with it `v0.4.1`? I need it to build pygcgopt.

I added the built files as `libscip-gcg*` in the `v0.4.0` release. We should probably make building with GCG controllable from the github workflow and have both versions with and without gcg in future releases.

**Pull request merged**